core/src/system_monitor_service.rs

```diff
@@ -690,6 +695,112 @@ impl SystemMonitorService {
         })
     }
 
+    /*
+    fn string_append_u32_as_char(&mut string: String, val: u32)
+    {
+        let mut v = val;
+
+        for _i in 0..4 {
+            string.push(v as u8 as char);
+            v >>= 8;
+        };
+    }*/
+
+    #[cfg(any(target_arch = "x86_64", target_arch = "x86_64"))]
+    unsafe fn report_cpuid_values() {
+        let cpuid_0 = __cpuid(0);
+
+        let max_leaf = cpuid_0.eax;
+        let max_subleaf_7 = if 7 <= max_leaf {
+            __get_cpuid_max(7).1
+        } else {
+            0
+        };
+
+        let mut manufacturer = String::from("");
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ebx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.edx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ecx >> (i * 8)) as u8 as char);
+        }
+
+        let cpuid_1 = if 1 <= max_leaf {
+            __cpuid(1)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
```

Comment:
lmao... no `Default` impl.  a constant would clean this up
```rust
const EMPTY_CPUID_RESULT: CpuidResult = CpuidResult { eax: 0, ebx: 0, ecx: 0, edx: 0 };
```

Reply:
Yeah, I very nearly did that, but.. no default. Why is there no default?  I can do that. Hopefully fmt won't make me expand it, but it probably will.